src/main/java/org/openrewrite/staticanalysis/MethodNameCasing.java

```diff
@@ -113,37 +104,9 @@ public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, Ex
                     !method.isConstructor() &&
                     !simpleName.startsWith("_") &&
                     !STANDARD_METHOD_NAME.matcher(simpleName).matches()) {
-                    StringBuilder standardized = new StringBuilder();
                     String normalized = VariableNameUtils.normalizeName(simpleName);
-
-                    if (SNAKE_CASE.matcher(normalized).matches()) {
-                        standardized.append(NameCaseConvention.format(NameCaseConvention.LOWER_CAMEL, normalized));
-                    } else {
-                        int nameLength = normalized.length();
-                        for (int i = 0; i < nameLength; i++) {
-                            char c = normalized.charAt(i);
-
-                            if (i == 0) {
-                                // the java specification requires identifiers to start with [a-zA-Z$_]
-                                if (c != '$' && c != '_') {
-                                    standardized.append(Character.toLowerCase(c));
-                                }
-                            } else {
-                                if (!Character.isLetterOrDigit(c)) {
-                                    while (i < nameLength && (!Character.isLetterOrDigit(c) || c > 'z')) {
-                                        c = normalized.charAt(i++);
-                                    }
-                                    if (i < nameLength) {
-                                        standardized.append(Character.toUpperCase(c));
-                                    }
-                                } else {
-                                    standardized.append(c);
-                                }
-                            }
-                        }
-                    }
-
-                    String toName = standardized.toString();
+                    NamingService service = getCursor().firstEnclosing(SourceFile.class).service(NamingService.class);
```

Comment:
The `JavaVisitor` in fact has a `service()` method which is much easier to use. We might want to pull that up to `TreeVisitor` at some point.

We might also have to cache the services a bit more as "breadcrumbs" along the cursor path, so that the service lookups don't get too expensive for deeply nested elements. But that is also something for another day.